week2/money-diary/main.js

```diff
@@ -0,0 +1,220 @@
+import { INIT_BALANCE, HISTORY_LIST } from "./data.js";
+
+let currentDeposit = INIT_BALANCE;
+let currentWithdraw = INIT_BALANCE;
+let currentBalance = INIT_BALANCE;
+
+const deposit = document.querySelector("b.deposit");
+const withdraw = document.querySelector("b.withdraw");
+const balance = document.getElementById("balance");
+
+const depositButton = document.querySelector(".deposit_input");
+const withdrawButton = document.querySelector(".withdraw_input");
+
+let currentHistoryList = HISTORY_LIST;
+
+const filter = () => {
+  currentHistoryList = [];
+
+  if (depositButton.checked) {
+    HISTORY_LIST.forEach((history) => {
+      if (history.type === "price_deposit") {
+        currentHistoryList.push(history);
+        renderHistory();
+      }
+    });
+  }
+  if (withdrawButton.checked) {
+    HISTORY_LIST.forEach((history) => {
+      if (history.type === "price_withdraw") {
+        currentHistoryList.push(history);
+        renderHistory();
+      }
+    });
+  }
+  if (!withdrawButton.checked && !depositButton.checked) {
+    renderHistory();
+  }
+};
```

Comment:
`renderHistory()`는 모든 currentHistoryList에 대한 push가 끝나고 딱 한 번만 돌려줘도 되는 친구 아닌가용?! 그게 맞다면
```suggestion
const filter = () => {
  currentHistoryList = [];

  if (depositButton.checked) {
    HISTORY_LIST.forEach((history) => {
      history.type === "price_deposit" && currentHistoryList.push(history);
    });
  }
  if (withdrawButton.checked) {
    HISTORY_LIST.forEach((history) => {
      history.type === "price_withdraw" && currentHistoryList.push(history);
    });
  }
  renderHistory();
};
```
이렇게 가능하겠네요 !!
AND 연산자 / if 연산자는 취향 ~!~